### PR TITLE
fix #1286: Expand/Collapse all buttons disappears for flat structures

### LINKED ISSUE #1286: Expand all and Collapse all should be disabled for flat structures
What if a case that my example or schema model doesn't contains a array and my object is pure single dimensional.

For example:
image

we should disable expand all,collapse all options for flat objects. **Expand All**, **Collapse All** doesn't work in some cases and above is best example. user may feel that something wrong and not working.

### PR DESCRIPTION
**Pull request by AvroraPolnareff**
Hi, I've done work on issue #1286. I think toggle only buttons visibility is easier way to implement this. Let me know if you prefer something else.

### CODE REVIEW

File: src/components/JsonViewer/JsonViewer.tsx [hunk 1]
```diff
@@ -26,27 +26,39 @@ class Json extends React.PureComponent<JsonProps> {
...
-            dangerouslySetInnerHTML={{
-              __html: jsonToHTML(this.props.data, options.jsonSampleExpandLevel),
-            }}
-          />
-        )}
-      </OptionsContext.Consumer>
-    </JsonViewerWrap>
-  );
+  renderInner = ({ renderCopyButton }) => {
+    const showFoldingButtons = Object.values(this.props.data).some(
```
Comment (RomanHotsiy) [inline src/components/JsonViewer/JsonViewer.tsx:30]:
```suggestion
    const showFoldingButtons = this.props.data && Object.values(this.props.data).some(
```

File: src/components/JsonViewer/JsonViewer.tsx [hunk 2]
```diff
@@ -26,27 +26,39 @@ class Json extends React.PureComponent<JsonProps> {
...
+      (value) => typeof value === 'object',
```
Comment (RomanHotsiy) [inline src/components/JsonViewer/JsonViewer.tsx:31]:
```suggestion
      (value) => typeof value === 'object' && value !== null,
```

File: src/components/__tests__/JsonViewer.tsx [hunk 3]
```diff
@@ -42,5 +42,16 @@ describe('Components', () => {
 
       expect(ClipboardService.copySelected as jest.Mock).toHaveBeenCalled();
     });
+
+    test('Expand/Collapse buttons disappears for flat structures', () => {
+      const flatData = { a: 1, b: '2' };
```
Comment (RomanHotsiy) [inline src/components/__tests__/JsonViewer.tsx:47]:
```suggestion
      const flatData = { a: 1, b: '2', c: null };
```

**Review by RomanHotsiy — CHANGES_REQUESTED** [review]
Suggested a few corrections to properly handle `null` values

File: src/components/JsonViewer/JsonViewer.tsx (hunk 2 above)
Comment (AvroraPolnareff) [reply src/components/JsonViewer/JsonViewer.tsx]: Thank for suggestion! I tested it, all works fine.

**Review by AvroraPolnareff — COMMENTED** [review]
All works fine, I've added suggestions to pull request.

**Comment by zalesky** [comment]
Screenshot 2022-04-22 at 11 48 08

LGTM

@AvroraPolnareff, Thanks for your input 👍

**Review by zalesky — APPROVED** [review]

**Comment by zalesky** [comment]
@RomanHotsiy  Would you kindly unblock this?

**Review by RomanHotsiy — APPROVED** [review]

**Pull request merged**